cocotb/utils.py

```diff
@@ -545,3 +545,123 @@ def remove_traceback_frames(tb_or_exc, frame_names):
             assert tb.tb_frame.f_code.co_name == frame_name
             tb = tb.tb_next
         return tb
+
+
+class SimTime():
+
+    def __init__(self, time, unit=None):
+        if isinstance(time, SimTime):
+            tme = time.steps
+        self.steps = get_sim_steps(time, unit)
+        self.unit = unit
```

Comment:
I'm not sure this property makes sense - when you get here, the time is now in "sim units"

Reply:
I guess it depends upon your point of view. Do operations on `SimTime` objects that underflow the simulator's precision even make sense? The current implementation throws if you try to create a duration less than the simulator's precision. If we go to a floating point implementation we have to worry about how to round to the simulator's precision. With this implementation we get floordiv and remainder so we don't lose any data. It's a design decision, is the underlying type abstract or concrete?

Reply:
I think we're talking past each other here. My complaint is that the `units` attribute doesn't actually tell me anything about the object. Changing it does not have any effect on the length of time it represents.

I also think line 554 is suspicious - to me `SimTime(SimTime(1, 'sec'), 'sec')` is nonsense, but for you it creates something that depends on the simulator resolution.

Reply:
The whole reason I have the units member is for serialization. I originally didn't store it, got to `__str__` and started scratching my head as to what a good representation was without units. But I'm not against just removing `__str__` and all references to units. Alternatively, you could think of units as a runtime type parameter.

Reply:
> started scratching my head as to what a good representation was without units

I might be tempted to just pick a unit value that results in a magnitude between 0 and 1000 (ie, engineering notation). You could add an overload of __format__ to let the user choose the unit for printing.